lib/internal/validators.js

```diff
@@ -35,24 +35,17 @@ function validateMode(value, name, def) {
   }
 
   if (typeof value === 'number') {
-    if (!Number.isInteger(value)) {
-      throw new ERR_OUT_OF_RANGE(name, 'an integer', value);
-    } else {
-      // 2 ** 32 === 4294967296
-      throw new ERR_OUT_OF_RANGE(name, '>= 0 && < 4294967296', value);
-    }
+    validateInt32(value, name, 0, 2 ** 32 - 1);
```

Comment:
do you know if we incur a cost doing this at runtime or does V8 compile it away? tbh I'd be more comfortable with a hardwired number here

Reply:
> do you know if we incur a cost doing this at runtime or does V8 compile it away? tbh I'd be more comfortable with a hardwired number here

I had the same thought but then figured that validating a mode passed to an `fs` function is never going to be in the hot path or a bottleneck in an `fs` operation, so performance impact should be negligible?

Reply:
I suppose so, still seems ripe for a constant though, `const MAX_INT32 = 2 ** 32 - 1;`

Reply:
@rvagg the number is constant fold by V8. That aside: this is already the error case. If it's a valid number, it would already be handled at this point. I just use the validation function to reduce code overhead.